source/opt/local_single_store_elim_pass.cpp

```diff
@@ -31,18 +31,22 @@ const uint32_t kStoreValIdInIdx = 1;
 
 bool LocalSingleStoreElimPass::HasOnlySupportedRefs(uint32_t ptrId) {
   if (supported_ref_ptrs_.find(ptrId) != supported_ref_ptrs_.end()) return true;
-  analysis::UseList* uses = get_def_use_mgr()->GetUses(ptrId);
-  assert(uses != nullptr);
-  for (auto u : *uses) {
-    SpvOp op = u.inst->opcode();
+  bool hasOnlySupportedRefs = true;
+  get_def_use_mgr()->ForEachUser(ptrId, [this,&hasOnlySupportedRefs](ir::Instruction* user) {
+    SpvOp op = user->opcode();
     if (IsNonPtrAccessChain(op) || op == SpvOpCopyObject) {
-      if (!HasOnlySupportedRefs(u.inst->result_id())) return false;
+      if (!HasOnlySupportedRefs(user->result_id())) {
+        hasOnlySupportedRefs = false;
```

Comment:
The logic has shifted somewhat here.  This now sets the flag to false, but does not cut the search short.  Intended?  Or am I just confused?

Reply:
Intended. The early return doesn't work with ForEachUser().

Reply:
Not confused.   I've debated in my head whether the ForEachUse(r) methods should return a bool to indicate "keep going".  Then an early exit is to return false.

I don't know if that makes the common use too cumbersome.  ?

Or should we have another method with the short-circuit.  But I don't have a good name for it: ForEachUseContinue
ForEachUseAnd

???  What's a good name?

I'm ok with doing that modification in a future CL.  Or if @dnovillo wants it now I'm cool with that too.

Reply:
I thought about supporting the early returns, but it felt awkward for most of the code I wanted to write.

I think having new methods is probably better than forcing every lambda to have "return true" at the end. In that case, we can keep the basic functionality with the same unambiguous name. I don't have any good suggestions for the new method name though.